frontend/src/metabase/admin/databases/containers/DatabaseEditApp.unit.spec.js

```diff
@@ -34,46 +37,16 @@ jest.mock(
   () => ComponentMock,
 );
 
-function mockSettings({ cachingEnabled = false }) {
-  const original = MetabaseSettings.get.bind(MetabaseSettings);
-  const spy = jest.spyOn(MetabaseSettings, "get");
-  spy.mockImplementation(key => {
-    if (key === "engines") {
-      return ENGINES_MOCK;
-    }
-    if (key === "enable-query-caching") {
-      return cachingEnabled;
-    }
-    if (key === "site-url") {
-      return "http://localhost:3333";
-    }
-    if (key === "application-name") {
-      return "Metabase Test";
-    }
-    if (key === "is-hosted?") {
-      return false;
-    }
-    if (key === "cloud-gateway-ips") {
-      return [];
-    }
-    return original(key);
-  });
-}
-
 async function setup({ cachingEnabled = false } = {}) {
-  mockSettings({ cachingEnabled });
-
-  const settingsReducer = () => ({
-    values: {
-      engines: ENGINES_MOCK,
-      "enable-query-caching": cachingEnabled,
-      "persisted-models-enabled": false,
-    },
+  const settings = mockSettings({
+    engines: ENGINES_MOCK,
+    "enable-query-caching": cachingEnabled,
+    "persisted-models-enabled": false,
   });
 
   renderWithProviders(<DatabaseEditApp />, {
     withRouter: true,
-    reducers: { admin, settings: settingsReducer },
+    reducers: { admin, settings: () => settings.state },
```

Comment:
we use only `initialState` in some tests, on the other hand, we use only `reducers` in some other tests. Is there a valid reason to have both ways to use `renderWithProviders`?

Reply:
Not really, #27552 will remove `reducers` and only keep `storeInitialState` option